internal/controller/api/management.go

```diff
@@ -155,6 +155,16 @@ func (s *ManagementServer) handleReconnect() http.HandlerFunc {
 			return
 		}
 
+		if reconnectReq.Delay < 0 {
+			errMsg := "Delay field cannot be negative"
```

Comment:
Eventually I would like to move the error handling logic (log the error, write an error response, etc) into a function but we do not necessarily have to do that now.

I would also like to move some of input validation logic into a function so that the main handler function is basically an outline of the logic of handling the request.  It would call the other functions where the details (input validation, etc) are handled.  I hope this makes sense.  We can discuss this idea over the phone if you like.  This does not have to be done now.